### PR TITLE
feat(containers/typeahead): new typeahead container

### PR DESCRIPTION
**Pull request by ncomont**
**What is the problem this PR is trying to solve?**

There is no container for the Typeahead component.

**What is the chosen solution to this problem?**

Add it.

**Please check if the PR fulfills these requirements**
- [x] The PR commit message follows our [guidelines](https://github.com/talend/tools/blob/master/tools-root-github/CONTRIBUTING.md)
- [x] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)
- [ ] Related design / discussions / pages (not in jira), if any, are all linked or available in the PR

**[ ] This PR introduces a breaking change**

### CODE REVIEW

File: packages/containers/src/Typeahead/Typeahead.container.js [hunk 1]
```diff
@@ -0,0 +1,42 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import Immutable from 'immutable';
+import { cmfConnect, componentState } from '@talend/react-cmf';
+import { Typeahead as Component } from '@talend/react-components';
+import omit from 'lodash/omit';
+
+export const DISPLAY_NAME = 'Container(TreeView)';
```
Comment (jmfrancois) [inline packages/containers/src/Typeahead/Typeahead.container.js]: ?

File: packages/containers/src/Typeahead/Typeahead.container.js [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+ * The Typeahead React container
+ */
+export default class Typeahead extends React.Component {
+	static displayName = DISPLAY_NAME;
+	static propTypes = {
+		// onClickActionCreator: PropTypes.string,
+		// onSelectActionCreator: PropTypes.string,
+		...componentState.propTypes,
+	};
+	// static defaultProps = DEFAULT_PROPS;
```
Comment (jmfrancois) [inline packages/containers/src/Typeahead/Typeahead.container.js]: remove those useless comments ;)

File: packages/containers/src/Typeahead/Typeahead.container.js (hunk 1 above)
Comment (frassinier) [reply packages/containers/src/Typeahead/Typeahead.container.js]: Wrong copy/paste for sure :)

File: packages/containers/src/Typeahead/Typeahead.connect.js [hunk 3]
```diff
@@ -0,0 +1,6 @@
+import { cmfConnect } from '@talend/react-cmf';
+import Container from './Typeahead.container';
+
+export default cmfConnect({
+	componentId: ownProps => ownProps.id || 'Typeahead',
```
Comment (frassinier) [inline packages/containers/src/Typeahead/Typeahead.connect.js]: Not a default value? ping @jmfrancois

File: packages/containers/src/Typeahead/Typeahead.container.js (hunk 2 above)
Comment (ncomont) [reply packages/containers/src/Typeahead/Typeahead.container.js]: That's just a draft, don't worry :)

File: packages/containers/src/Typeahead/Typeahead.container.js [hunk 4]
```diff
@@ -0,0 +1,127 @@
...
+import { cmfConnect, componentState } from '@talend/react-cmf';
+import { Typeahead as Component } from '@talend/react-components';
+import omit from 'lodash/omit';
+
+const KEYS = {
+	DOWN: 'ArrowDown',
+	UP: 'ArrowUp',
+	ENTER: 'Enter',
+	ESC: 'Escape',
+};
```
Comment (jmfrancois) [inline packages/containers/src/Typeahead/Typeahead.container.js]: please remove this and use keycode (it already be a dependency)

File: packages/containers/src/index.js [hunk 5]
```diff
@@ -6,7 +6,6 @@ import {
 	IconsProvider,
 	Layout,
 	TooltipTrigger,
-	Typeahead,
```
Comment (jmfrancois) [inline packages/containers/src/index.js]: 👍

File: packages/containers/src/Typeahead/Typeahead.container.js [hunk 6]
```diff
@@ -0,0 +1,120 @@
...
+		const props = {
+			...omit(this.props, cmfConnect.INJECTED_PROPS),
+			onToggle: this.onToggle,
+			onBlur: this.onBlur,
+			onSelect: this.onSelect,
+			onKeyDown: this.onKeyDown,
+			docked: state.get('docked'),
+			searching: state.get('searching'),
+			focusedSectionIndex: state.get('focusedSectionIndex'),
+			focusedItemIndex: state.get('focusedItemIndex'),
```
Comment (jmfrancois) [inline packages/containers/src/Typeahead/Typeahead.container.js]: please use ...state.toJS() instead of the four previous lines

**Pull request merged**